version/derive/src/lib.rs

```diff
@@ -179,59 +168,15 @@ fn impl_version(
             "Versioned struct name argument should have a string value."
         )
     };
-    let wrapper_struct_name = format_ident!("_{}V{}", versioned_struct_name, version_number);
-    let versioned_struct_name = Ident::new(&versioned_struct_name, Span::call_site());
-    let json_derives = if with_json {
-        quote!(serde::Serialize, serde::Deserialize,)
-    } else {
-        quote!()
-    };
-    let scale_derives = if with_scale {
-        quote!(parity_scale_codec::Encode, parity_scale_codec::Decode,)
-    } else {
-        quote!()
-    };
+    let alias_type_name = format_ident!("_{}V{}", versioned_struct_name, version_number);
     quote!(
-        /// Autogenerated wrapper struct to link versioned item to its container.
-        #[derive(#scale_derives #json_derives #derives)]
-        pub struct #wrapper_struct_name (#struct_name);
-
-        impl From<#wrapper_struct_name> for #struct_name {
-            fn from(wrapper: #wrapper_struct_name) -> Self {
-                let #wrapper_struct_name (inner) = wrapper;
-                inner
-            }
-        }
-
-        impl From<#struct_name> for #wrapper_struct_name {
-            fn from(origin: #struct_name) -> Self {
-                #wrapper_struct_name (origin)
-            }
-        }
-
-        impl From<#struct_name> for #versioned_struct_name {
-            fn from(origin: #struct_name) -> Self {
-                let wrapped: #wrapper_struct_name = origin.into();
-                wrapped.into()
-            }
-        }
+        /// Autogenerated alias type to link versioned item to its container.
+        type #alias_type_name = #struct_name;
```

Comment:
Does type alias really provide versioning?
To me, for example, v1 and v2 seem to be the same type

Reply:
I see versioning is provided by `enum Versioned*`.
So previously it had double wrapping